go/testimonyd/internal/socket/socket.c

```diff
@@ -28,6 +28,26 @@
 #define UNIX_PATH_MAX 108
 #endif
 
+// Function for waiting for new blocks using select() function
+int WaitForBlocks(int sock_fd) {
```

Comment:
document return value. 

Also consider returning errno on failure, for use with syscall.Errno

Reply:
changed return value to 0 for success (timeout or ready) and errno if select returns -1.